component/attr_cache/attr_cache.go

```diff
@@ -556,11 +550,13 @@ func (ac *AttrCache) DeleteFile(options internal.DeleteFileOptions) error {
 		ac.cacheLock.RLock()
 		defer ac.cacheLock.RUnlock()
 		toBeDeleted, getErr := ac.cacheMap.get(options.Name)
-		if getErr != nil {
-			log.Err("AttrCache::DeleteFile : %s", getErr)
-		} else {
-			toBeDeleted.markDeleted(deletionTime)
+		if getErr != nil || !toBeDeleted.valid() {
+			log.Warn("AttrCache::DeleteFile : %s no valid entry found. Adding entry...", options.Name)
+			// add deleted file entry
+			attr := internal.CreateObjAttr(options.Name, 0, deletionTime)
```

Comment:
What are the benefits of adding the item to the cache right before marking it as deleted?

Reply:
The attribute cache needs to keep a record of the fact that this file is deleted.
I thought about passing `false` for exists here, but it seemed slightly more readable to:

1. Find or create the entry
2. Mark it deleted